plugins/woocommerce/src/Blocks/BlockTypes/AddtoCartWithOptions/Utils.php

```diff
@@ -0,0 +1,128 @@
+		while ( $html->next_tag( array( 'class_name' => 'input-text' ) ) ) {
+			$html->add_class( 'wc-block-components-quantity-selector__input' );
+		}
+
+		return $html->get_updated_html();
+	}
+
+	/**
+	 * Get standardized quantity input arguments for WooCommerce quantity input.
+	 *
+	 * @param \WC_Product $product The product object.
+	 * @return array Arguments for woocommerce_quantity_input().
+	 */
+	public static function get_quantity_input_args( $product ) {
+		return array(
+			/**
+			 * Filter the minimum quantity value allowed for the product.
+			 *
+			 * @since 2.0.0
+			 * @param int        $min_value Minimum quantity value.
+			 * @param WC_Product $product   Product object.
+			 */
+			'min_value'   => apply_filters( 'woocommerce_quantity_input_min', $product->get_min_purchase_quantity(), $product ),
+			/**
+			 * Filter the maximum quantity value allowed for the product.
+			 *
+			 * @since 2.0.0
+			 * @param int        $max_value Maximum quantity value.
+			 * @param WC_Product $product   Product object.
+			 */
+			'max_value'   => apply_filters( 'woocommerce_quantity_input_max', $product->get_max_purchase_quantity(), $product ),
+			'input_value' => isset( $_POST['quantity'] ) ? wc_stock_amount( wp_unslash( $_POST['quantity'] ) ) : $product->get_min_purchase_quantity(), // phpcs:ignore WordPress.Security.NonceVerification.Missing
+		);
+	}
+
+	/**
+	 * Make the quantity input interactive by wrapping it with the necessary data attribute.
+	 *
+	 * @param string $quantity_html The quantity HTML.
+	 * @param string $wrapper_attributes Optional wrapper attributes.
+	 * @return string The quantity HTML with interactive wrapper.
+	 */
+	public static function make_quantity_input_interactive( $quantity_html, $wrapper_attributes = '' ) {
+		if ( ! empty( $wrapper_attributes ) ) {
+			return sprintf(
+				'<div %1$s data-wp-interactive="woocommerce/add-to-cart-with-options">%2$s</div>',
+				$wrapper_attributes,
+				$quantity_html
+			);
+		}
+
+		return '<div data-wp-interactive="woocommerce/add-to-cart-with-options">' . $quantity_html . '</div>';
+	}
+
+	/**
+	 * Get product from block context, preserving the global product.
+	 *
+	 * @param \WP_Block   $block The block instance.
+	 * @param \WC_Product $previous_product The previous product (usually from global scope).
+	 * @return array Array containing the product to use and a boolean indicating if a valid product was found.
+	 */
+	public static function get_product_from_context( $block, $previous_product ) {
+		$post_id = isset( $block->context['postId'] ) ? $block->context['postId'] : '';
+		$product = null;
+
+		if ( ! empty( $post_id ) ) {
+			$product = wc_get_product( $post_id );
+		}
+
+		if ( ! $product instanceof \WC_Product && $previous_product instanceof \WC_Product ) {
+			$product = $previous_product;
+		}
+
+		$is_valid_product = $product instanceof \WC_Product;
+
+		return array(
+			'product'  => $product,
+			'is_valid' => $is_valid_product,
+		);
```

Comment:
I would prefer if this method only returned `$product`. If the product isn't valid, we can simply return `null`. Something like this:

```suggestion
		return $product instanceof \WC_Product ? $product : null;
```

You will need to update the functions where this method is consumed. But IMO it's better to keep the signature of the method simpler.

Reply:
Done and fixed the linting error.